test/org/beanio/types/NumberTypeHandlerTest.java

```diff
@@ -75,4 +75,12 @@ public void testInvalidPattern() {
         IntegerTypeHandler handler = new IntegerTypeHandler();
         handler.setPattern("0.00.00");
     }
+
+    @Test
+    public void testDifferentThreads() {
+        IntegerTypeHandler handler = new IntegerTypeHandler();
+        handler.setPattern("#");
+        handler.format(10);
+        new Thread(() -> {handler.format(10);}).start();

```

Comment:
If `handler.format(10)` throws, the exception won't be propagated to `testDifferentThreads` and JUnit won't catch it. A better option would be to assign the result to a variable and make an assertion:
```java
	@Test
	public void testDifferentThreads() throws InterruptedException {
		IntegerTypeHandler handler = new IntegerTypeHandler();
		handler.setPattern("#");

		assertEquals("10", handler.format(10));

		CountDownLatch latch = new CountDownLatch(1);
		AtomicReference<String> value = new AtomicReference<>();

		new Thread(() -> {
			value.set(handler.format(10));
			latch.countDown();
		}).start();

		latch.await(5, TimeUnit.SECONDS);

		assertEquals("10", value.get());
	}
```

Reply:
true, thanks!, submitted a fix with a bit less verbose approach